nixos/modules/system/boot/systemd.nix

```diff
@@ -451,6 +451,21 @@ in
         cfg.services
     );
 
+    assertions = concatLists (
+      mapAttrsToList
+        (name: service:
+          map (message: {
+            assertion = false;
+            inherit message;
+          }) (concatLists [
```

Comment:
This assertion is very confusingly written. Why is there a `concatLists` for a list with a single element? Why is it mapping a list made with `optional` to an assertion with `assertion = false` instead of just putting the condition in the `assertion` field?

Reply:
> assertion = false

I don't know the impact of adding a lot of assertions (as there would be one for every system service)

> concatLists

This is copied from the warnings above, to support multiple assertions later.

Reply:
Any further objections? @ElvishJerricco

Reply:
I still think this was extremely weirdly written and should have been cleaned up, but I wasn't going to block the PR on it